File: kyonggiyo-core/src/main/java/kyonggiyo/application/service/auth/OAuthQueryService.java
package kyonggiyo.application.service.auth;

import kyonggiyo.application.port.out.auth.LoadOAuthTokenPort;
import kyonggiyo.application.port.out.auth.LoadOAuthUserInfoPort;
import kyonggiyo.domain.auth.Platform;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class OAuthQueryService {

    private final LoadOAuthTokenPort loadOAuthTokenPort;
    private final LoadOAuthUserInfoPort loadOAuthUserInfoPort;

    public String getProviderId(Platform platform, String authCode) {
        String accessToken = loadOAuthTokenPort.requestToken(platform, authCode);
        return loadOAuthUserInfoPort.requestUserInfo(platform, accessToken);
    }
    
}

File: kyonggiyo-core/src/main/java/kyonggiyo/application/auth/service/OAuthQueryService.kt
package kyonggiyo.application.auth.service

import kyonggiyo.application.auth.domain.vo.Platform
import kyonggiyo.application.auth.port.outbound.LoadOAuthTokenPort
import kyonggiyo.application.auth.port.outbound.LoadOAuthUserInfoPort
import org.springframework.stereotype.Service

@Service
class OAuthQueryService(
        private val loadOAuthTokenPort: LoadOAuthTokenPort,
        private val loadOAuthUserInfoPort: LoadOAuthUserInfoPort
) {

    fun getProviderId(platform: Platform, authCode: String): String {
        return loadOAuthTokenPort.requestToken(platform, authCode)
                .let { loadOAuthUserInfoPort.requestUserInfo(platform, it) }
    }

}
